src/Stratis.Bitcoin/Consensus/ChainedHeaderTree.cs

```diff
@@ -446,8 +448,12 @@ public List<int> ConsensusTipChanged(ChainedHeader newConsensusTip)
             // Consider blocks that became a part of our best chain as consumed.
             while (currentHeader != fork)
             {
-                this.UnconsumedBlocksDataBytes -= currentHeader.Block.BlockSize.Value;
-                this.logger.LogTrace("Size of unconsumed block data is decreased by {0}, new value is {1}.", currentHeader.Block.BlockSize.Value, this.UnconsumedBlocksDataBytes);
+                if (currentHeader.BlockDataAvailability == BlockDataAvailabilityState.BlockAvailable)
```

Comment:
@noescape00 @Aprogiena @dangershony Can you please comment on this? I ran into a null reference for `currentHeader.Block`  for the block just after the fork point. It was full validated but the block was null and `BlockDataAvailabilityState` was `HeaderOnly`.

Reply:
pls create issue for that

Reply:
This fixes the issue? Or do you just want to have it for our records?

Reply:
Do you have a way to reproduce this constantly?
You are fixing more then one issue in one PR this makes it harder to be sure a fix is correct, or if a fix needs to be rolled back etc...

Reply:
@dangershony I could not advance after this without this fix, this is why I asked your thoughts on this as I am unsure why `block` would be null here.

Reply:
I am not sure myself I will need to reproduce somehow, I got the same bug but after a restart I could continue to sync

Reply:
This is much more complicated, because this should never happen so that is why we believe it only happens because of the first bug. If that is the case, there is nothing to fix here. If that is not the case, it needs much deeper look, because this does not fix it. But we first need to fix the first bug and then to see if there is another one.

Reply:
i can't find anything relevant to it, can you link or copy paste please?

Reply:
or do you mean https://github.com/stratisproject/StratisBitcoinFullNode/pull/2216#issuecomment-422406237 ?

Reply:
if yes, that should not happen and if it does we should investigate separately, that is what i'm explaining above